src/kOS/Execution/Stack.cs

```diff
@@ -48,9 +52,19 @@ private void ThrowIfInvalid(object item)
         /// <returns></returns>
         private object ProcessItem(object item)
         {
-            if (item is float)
+            if (item is float) {
                 // promote floats to doubles
-                return Convert.ToDouble(item);
+                item = Convert.ToDouble (item);
+            }
+            if (item is double)
+            {
+                if (!Double.IsNaN((double)item) && Int32.MinValue < (double)item && (double)item < Int32.MaxValue)
+                {
+                    int intPart = Convert.ToInt32(item);
+                    if ((double)item == (double)intPart)
+                        item = intPart;
+                }
+            }
```

Comment:
@Dunbaratu i was more worried about floating point error making many of the ints not match if is has been a part of any math operation.

it still seems pretty heavy for something we do each time we put something on the stack

it seems like either of these would be cheaper to use for the check

    double d = 5.0;
    bool isInt = d == (int)d;

    double d = 5.0;
    bool isInt = d % 1 == 0;

but if you guys are ok with it maybe im just worried over nothing :)